bc_obps/service/data_access_service/operation_service.py

```diff
@@ -64,14 +64,46 @@ def create_operation(
         operation_data: DictStrAny,
         regulated_products: Union[QuerySet[RegulatedProduct], Iterable[RegulatedProduct]],
     ) -> Operation:
-
         operation = Operation.objects.create(
             **operation_data,
             created_by_id=user_guid,
         )
         operation.regulated_products.set(regulated_products)
+
+        return operation
+
+    @classmethod
+    def create_operation_v2(
+        cls,
+        user_guid: UUID,
+        operation_data: DictStrAny,
+        activities: Union[QuerySet[Activity], Iterable[Activity]],
+    ) -> Operation:
+        operation = Operation.objects.create(
+            **operation_data,
+            created_by_id=user_guid,
+        )
+        operation.activities.set(activities)
+
```

Comment:
I thiiiink we need to call `operation.set_create_or_update()` here in L87, but want to double-check with @Sepehr-Sobhani because frankly I've forgotten how to write code.

Reply:
Since we are creating an operation here and explicitly passing `created_by_id`, we no longer need to call `set_create_or_update()`. However, if `operation.activities.set(activities)` were outside of this service, we should have called it because we are actually performing an update on the operation instance.